File: typescript/src/demo/scenario.ts
import { AuditHost } from '../host/auditHost.js';
import { InProcessTransport } from '../transport/inProcess.js';
import { AmcpSession, deterministicDeps } from '../tool/amcp.js';
import { ResearchTool } from '../tool/researchTool.js';

// A reproducible L1 scenario shared by the demo, the tests, and the chain conformance vector.
// A first-party research tool performs three internal operations that deliberately span the
// (mutates, egress) axis: a web search (read-only, yet the query egresses), a note write, and
// a note read.
export async function runCleanScenario(partition = 'acme#2026-07-15'): Promise<AuditHost> {
  const host = new AuditHost(partition);
  const transport = new InProcessTransport(host);
  const session = new AmcpSession(transport, 'call_abc', deterministicDeps());
  const tool = new ResearchTool(session);

  await tool.search('acme corp merger due diligence'); // api.request — mutates=0, egress=1
  await tool.saveNote('acme', 'merger rumour confirmed by two sources'); // db.write
  await tool.listNotes(); // db.read

  return host;
}

File: python/src/a_mcp/demo/scenario.py
'''A reproducible L1 scenario shared by the demo and the tests.

A first-party customer DB tool performs a few internal reads/writes; every operation is
self-attested and sealed.
'''

from a_mcp.amcp import AmcpSession, DeterministicDeps
from a_mcp.customer_db_tool import CustomerDbTool
from a_mcp.host import AuditHost
from a_mcp.in_process import InProcessTransport


def run_clean_scenario(partition: str = 'acme#2026-07-15') -> AuditHost:
    '''Run the clean scenario and return the host holding the sealed ledger.'''
    host = AuditHost(partition)
    session = AmcpSession(InProcessTransport(host), 'call_abc', DeterministicDeps())
    tool = CustomerDbTool(session)
    tool.get_customer('c_1')
    tool.update_email('c_1', 'new@acme.example')
    tool.get_customer('c_2')
    return host
    # end def
